molecule/core.py

```diff
@@ -198,14 +198,33 @@ def _create_templates(self):
 
         # rakefile
         kwargs = {
-            'molecule_file': self._config.molecule_file,
-            'current_platform': self._provisioner._platform,
+            'state_file': self._config.config['molecule']['state_file'],
             'serverspec_dir': self._config.config['molecule']['serverspec_dir']
         }
         utilities.write_template(
             self._config.config['molecule']['rakefile_template'],
             self._config.config['molecule']['rakefile_file'], kwargs=kwargs)
 
+    def _create_instances_state(self):
+        """
+        Creates a dict of formatted instances names and the group(s) they're
+        part of to be added to state.
+
+        :return: None
+        """
+
+        instances = {}
+        for instance in self._provisioner.instances:
+            instance_name = utilities.format_instance_name(
+                instance['name'], self._provisioner._platform,
+                self._provisioner.instances)
+            instances[instance_name] = []
+            if 'ansible_groups' in instance:
+                for group in instance['ansible_groups']:
+                    instances[instance_name].append(group)
+
+        self._state.change_state('instances', instances)
```

Comment:
The issue is that the rakefile needs to know the instance names. We used to be able to infer that from the molecule.yml file, because the names were either as named in the file, or with the platform appended, which is also an option in that file.  That's no longer the case since `--platform all` automatically appends the platform name to a hostname. That's why this was broken. The rakefile was trying to hit demo-01-all.

So we need a way to pass that information to the Rakefile now. Ruby can't easily parse the actual inventory file, so the state file seemed like the next best choice since it's YML, and the actual instance names being used are now part of the state of molecule while they're running.